Unicode/ustring.c

```diff
@@ -301,26 +301,36 @@ return( NULL );
 
 unichar_t *u_copyn(const unichar_t *pt, long n) {
     unichar_t *res;
+
+    if ( pt && ++n>0 ) {
 #ifdef MEMORY_MASK
-    if ( n*sizeof(unichar_t)>=MEMORY_MASK )
-	n = MEMORY_MASK/sizeof(unichar_t)-1;
+	if ( n>MEMORY_MASK/sizeof(unichar_t) )
```

Comment:
can you explain this one a bit more

Reply:
And perhaps add a comment indicating what MEMORY_MASK is. The name doesn't seem to pertain to the use.

Reply:
Looking lower down in the code, we need to build for malloc((n+1)*sizeof(unichar_t)), so we still need room for one more additional sizeof(unichar_t).
Moving the n+1 up to the top avoids several repeated computations....
and moving the sizeof(unichar_t) to the other side of the equation lets the compiler precompute the value once instead of all the users computing it repeatedly during run time.

Reply:
Reviewing the MEMORY_MASK using git blame (on a copy of 20120731 code), seems this code came in around 2001 and seems to be nowhere else other than within ustring.h
Maybe an abandoned idea?
Take it out completely?

Reply:
@JoesCat, n will be in a register in any reasonably compiled binary, and it's not even in the inner loop. So all of this might save two clock cycles per string copy. I don't think that that justifies the significant decrease in readability.